Entities/LobbyMapController.cs

```diff
@@ -0,0 +1,319 @@
+            
+            /// <summary>
+            /// The type of marker. This allows for filtering in the controller.
+            /// </summary>
+            public MarkerType Type;
+            
+            /// <summary>
+            /// The position of the marker on the map.
+            /// </summary>
+            public Vector2 Position;
+            
+            /// <summary>
+            /// The SID for the lobby map.
+            /// </summary>
+            public string SID;
+            
+            /// <summary>
+            /// The room within the map that this marker belongs to.
+            /// </summary>
+            public string Room;
+            
+            /// <summary>
+            /// The name of the map to load when using a marker type of <see cref="MarkerType.Map"/>.
+            /// </summary>
+            public string Map;
+            
+            /// <summary>
+            /// Extracted data about the referenced map if it exists.
+            /// </summary>
+            public MapInfo MapInfo;
+
+            public static bool TryParse(EntityData data, ControllerInfo controllerInfo, out MarkerInfo value) {
+                value = default;
+                
+                // CU2 simple marker entity
+                if (data.Name == "CollabUtils2/LobbyMapMarker") {
+                    value.Type = MarkerType.Custom;
+                    value.DialogKey = data.Attr("dialogKey");
+                    value.Icon = data.Attr("icon");
+                }
+                // CU2 warp entity
+                else if (data.Name == "CollabUtils2/LobbyMapWarp") {
+                    value.Type = MarkerType.Warp;
+                    value.DialogKey = data.Attr("dialogKey");
+                    value.MarkerId = data.Attr("warpId");
+                    value.Icon = data.Attr("icon");
+                }
+                // CU2 rainbow berry
+                else if (data.Name == "CollabUtils2/RainbowBerry") {
+                    value.Type = MarkerType.RainbowBerry;
+                }
+                // CU2 heart door
+                else if (data.Name == "CollabUtils2/MiniHeartDoor") {
+                    value.Type = MarkerType.HeartDoor;
+                }
+                // CU2 journal trigger
+                else if (data.Name == "CollabUtils2/JournalTrigger") {
+                    value.Type = MarkerType.Journal;
+                }
+                // CU2 map trigger or something from the CustomMarkers property
+                else if (data.Name == "CollabUtils2/ChapterPanelTrigger" || controllerInfo != null && controllerInfo.CustomMarkers.Contains(data.Name)) {
+                    value.Map = data.Attr("map");
+                    value.Type = value.Map.Contains("0-Gyms") ? MarkerType.Gym : MarkerType.Map;
+                }
+                // XaphanHelper warp station (can only be warped to, not from)
+                else if (data.Name == "XaphanHelper/WarpStation" && controllerInfo != null) {
```

Comment:
I still think you shouldn't add support for Xaphan warps. the systems don't really integrate together (esp if people try to use multiple map types), and if XH changes anything it'll get messy

Reply:
Yeah fair call.